docs/data-type-support.rst

```diff
@@ -0,0 +1,86 @@
+.. meta::
+  :description: rocThrust API data type support
+  :keywords: rocThrust, ROCm, API, reference, data type, support
+
+.. _data-type-support:
+
+******************************************
+Data type support
+******************************************
+
+Fundamental types
+=================
+
+
+ .. list-table:: Comparison of supported fundamental types of rocThrust and Thrust
+    :header-rows: 1
+    :name: supported-fundamental-types-rocthrust-vs-thrust
+
+    *
+      - Type
+      - Size of type
+      - rocThrust support
+      - Thrust support
+    *
+      - :code:`unsigned char`
+      - 8 bit
+      - ✅
+      - ✅
+    *
+      - :code:`unsigned short`
+      - 16 bit
+      - ✅
+      - ✅
+    *
+      - :code:`unsigned int`
+      - 32 bit
+      - ✅
+      - ✅
+    *
+      - :code:`unsigned long long`
+      - 64 bit
+      - ✅
+      - ✅
+    *
+      - :code:`char`
+      - 8 bit
+      - ✅
+      - ✅
+    *
+      - :code:`short`
+      - 16 bit
+      - ✅
+      - ✅
+    *
+      - :code:`int`
+      - 32 bit
+      - ✅
+      - ✅
+    *
+      - :code:`long long`
+      - 64 bit
+      - ✅
+      - ✅
+    *
+      - :code:`half` [1]_
```

Comment:
I have a quick question about the footnote here. In the rocPRIM data type documentation table, I noticed that we currently don't support calling rocprim::device_reduce_by_key with the rocprim::half type. Does that mean we also don't support thrust::reduce_by_key with the half type on the device? If so, it might be nice to add to the footnote.

Reply:
I think `rocprim::device_reduce_by_key` actually fails due to verification problems on the host side. As the block/thread level functions work, I would expect it to also work in thrust. I can double check, but the verification would most likely also fail due to the missing functionality of `half` on the host